src/Agent.Listener/CommandSettings.cs

```diff
@@ -436,117 +453,230 @@ public string GetWork()
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public bool IsHelp()
+        {
+            if ((Configure != null && Configure.Help) ||
+                (Remove != null && Remove.Help) ||
+                (Run != null && Run.Help) ||
+                (Warmup != null && Warmup.Help))
+            {
+                return true;
+            }
+
+            return false;
         }
 
-        private void RemoveArg(string name)
+        public bool IsCommit()
         {
-            if (_parser.Args.ContainsKey(name))
+            if (Run != null && Run.Commit)
             {
-                _parser.Args.Remove(name);
+                return true;
             }
 
-            if (_envArgs.ContainsKey(name))
+            return false;
+        }
+
+        public bool IsRunDiagnostics()
+        {
+            if (Run != null && Run.Diagnostics)
             {
-                _envArgs.Remove(name);
+                return true;
             }
+
+            return false;
+        }
+
+        public bool IsConfigureCommand()
+        {
+            if (Configure != null)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public bool IsRemoveCommand()
+        {
+            if (Remove != null)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public bool IsWarmupCommand()
+        {
+            if (Remove != null)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+
+        //
+        // Private helpers.
+        //
+        private string GetArg(string value, string envName)
+        {
+            if (value == null)
+            {
+                value = GetEnvArg(envName);
+            }
+
+            return value;
         }
 
         private string GetArgOrPrompt(
+            string argValue,
             string name,
             string description,
             string defaultValue,
             Func<string, bool> validator)
         {
             // Check for the arg in the command line parser.
             ArgUtil.NotNull(validator, nameof(validator));
-            string result = GetArg(name);
+            string result = GetArg(argValue, name);
 
             // Return the arg if it is not empty and is valid.
             _trace.Info($"Arg '{name}': '{result}'");
             if (!string.IsNullOrEmpty(result))
             {
-                // After read the arg from input commandline args, remove it from Arg dictionary,
-                // This will help if bad arg value passed through CommandLine arg, when ConfigurationManager ask CommandSetting the second time,
-                // It will prompt for input instead of continue use the bad input.
-                _trace.Info($"Remove {name} from Arg dictionary.");
-                RemoveArg(name);
```

Comment:
While i could replicate this logic block, I don't see why it's worth doing.  The results of `GetArgOrPrompt` don't appear to be saved anywhere.  Hence if called twice you'd still enter the block below.